### PR TITLE
add Delete methods

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/2023

### CODE REVIEW

File: service/resource/kmskeyv1/delete.go [hunk 1]
```diff
@@ -3,13 +3,76 @@ package kmskeyv1
...
+			KeyId: keyAlias,
+		})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		// AWS API doesn't allow to delete the KMS key immediately, but we can schedule its deletion
+		if _, err := r.awsClients.KMS.ScheduleKeyDeletion(&kms.ScheduleKeyDeletionInput{
+			KeyId:               key.KeyMetadata.KeyId,
+			PendingWindowInDays: aws.Int64(7),
```
Comment (rossf7) [inline service/resource/kmskeyv1/delete.go]: Minor but I'd like to have a constant for this.

File: service/resource/kmskeyv1/create_test.go [hunk 2]
```diff
@@ -73,7 +69,7 @@ func Test_Resource_S3Object_newCreate(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
-		result, err := newResource.newCreateChange(context.TODO(), tc.obj, tc.currentState, tc.desiredState)
+		result, err := newResource.newCreateChange(context.TODO(), clusterTpo, tc.currentState, tc.desiredState)
```
Comment (rossf7) [inline service/resource/kmskeyv1/create_test.go]: Minor but can this be customObject? TPOs are going soon :)

File: service/resource/kmskeyv1/delete_test.go [hunk 3]
```diff
@@ -1 +1,143 @@
...
+	resourceConfig := DefaultConfig()
+	resourceConfig.Clients = Clients{}
+	resourceConfig.Logger = microloggertest.New()
+	newResource, err = New(resourceConfig)
+	if err != nil {
+		t.Error("expected", nil, "got", err)
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
```
Comment (rossf7) [inline service/resource/kmskeyv1/delete_test.go]: I really like that we're using subtests here. We should use them more widely.

File: service/resource/kmskeyv1/create_test.go (hunk 2 above)
Comment (fgimenez) [reply service/resource/kmskeyv1/create_test.go]: \o/

File: service/resource/kmskeyv1/delete.go (hunk 1 above)
Comment (fgimenez) [reply service/resource/kmskeyv1/delete.go]: Sure, done

File: service/resource/kmskeyv1/delete_test.go (hunk 3 above)
Comment (fgimenez) [reply service/resource/kmskeyv1/delete_test.go]: Indeed, I try to use them as much as i can :)

**Pull request merged**